joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/ControlStructureTests.scala

```diff
@@ -0,0 +1,132 @@
+import overflowdb.traversal.toNodeTraversal
+import io.joern.dataflowengineoss.language.toExtendedCfgNode
+import io.joern.dataflowengineoss.semanticsloader.FlowSemantic
+
+class ControlStructureTests extends RubyCode2CpgFixture {
+
+  "while-loop use cases 1" should {
+    val cpg = code("""
+        |def method()
+        |  counter = 0
+        |  while counter < 5 do
+        |    puts "Counter: #{counter}"
+        |    counter += 1
+        |  end
+        |end
+        |""".stripMargin)
+
+    // TODO: Need to be fixed.
+    "be correct for while loop" ignore {
+      cpg.method.name("method").block.astChildren.isControlStructure.l.size shouldBe 1
+      inside(cpg.method.name("method").block.astChildren.isControlStructure.l) {
+        case List(controlStruct: ControlStructure) =>
+          controlStruct.controlStructureType shouldBe ControlStructureTypes.WHILE
+          inside(controlStruct.condition.l) { case List(cndNode) =>
+            cndNode.code shouldBe "counter < 5"
+          }
+          controlStruct.whenTrue.astChildren.isCall.l.size should be > 2
+      }
+    }
+    // TODO: Need to be fixed. I think above unit test is cause of this data flow to not work
+    "data flow test" ignore {
+      val src  = cpg.identifier("counter").lineNumber(3).l
+      val sink = cpg.call("puts").l
+      sink.reachableByFlows(src).size shouldBe 1
+    }
+  }
+  "while-loop use cases 2" should {
+    val cpg = code("""
+        |def method()
+        |  counter = 0
+        |  begin
+        |    puts "Counter: #{counter}"
+        |    counter += 1
+        |  end while counter < 5
+        |end
+        |""".stripMargin)
+
+    // TODO: Need to be fixed
+    "data flow test" ignore {
+      val src  = cpg.identifier("counter").lineNumber(3).l
+      val sink = cpg.call("puts").l
+      sink.reachableByFlows(src).size shouldBe 1
+    }
+  }
+
+  "until-loop use cases 1" should {
+    val cpg = code("""
+        |def method()
+        |  counter = 0
+        |  until counter >= 5 do
+        |    puts "Counter: #{counter}"
+        |    counter += 1
+        |  end
+        |end
+        |""".stripMargin)
+
+    // TODO: Need to be fixed
+    "data flow test" ignore {
+      val src  = cpg.identifier("counter").lineNumber(3).l
+      val sink = cpg.call("puts").l
+      sink.reachableByFlows(src).size shouldBe 1
+    }
+  }
+
+  "until-loop use cases 2" should {
+    val cpg = code("""
+        |def method()
+        |  counter = 0
+        |  begin
+        |    puts "Counter: #{counter}"
+        |    counter += 1
+        |  end until counter < 5
+        |end
+        |""".stripMargin)
+
+    // TODO: Need to be fixed
+    "data flow test" ignore {
+      val src  = cpg.identifier("counter").lineNumber(3).l
+      val sink = cpg.call("puts").l
+      sink.reachableByFlows(src).size shouldBe 1
+    }
+  }
+
+  "if statement use case 1" should {
+    val cpg = code("""
+        |def method()
```

Comment:
This test is incorrect since there is no invocation of `method()`. The correct test case would be
```
def method()
        counter = 5 
        if counter > 1 
        puts "Counter: #{counter}"
        end 
end

method
```

And it works.

Please make sure the test has valid Ruby code that works on the shell

Reply:
I tried as per your suggestion test are still failing.